### PR TITLE
docs: how to create a new component

### PR DESCRIPTION
**Pull request by cuinixam**

### CODE REVIEW

File: docs/getting_started/start_new_component.rst [hunk 1]
```diff
@@ -0,0 +1,403 @@
+How to Start Working on a :ref:`Component <what_is_a_component>`
+################################################################
+
+To support :ref:`component <what_is_a_component>` development, SPL Core provides features to:
+
+- document the component behavior
+- test the component behavior
+- generate component report with full traceability between documentation, code and tests
+- **all** component artifacts are configurable (content will be included/excluded based on configuration)
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst]: follow the grammatic of the bullet points above and rephrase the sentence, like "configure **all** component artifacts [...]

File: docs/getting_started/start_new_component.rst [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+In order to trace the documentation to the code or tests, you need to add a `sphinx-needs <https://www.sphinx-needs.com/>`_ element.
+
+.. code-block:: rst
+
+    .. spec:: Feature X
+        :id: SWDD_COMP-001
+
+        The component shall implement feature X.
+
+The ``spec`` is used to tag the component specification. It shall have:
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst]: The ``spec`` keyword [...]

File: docs/getting_started/start_new_component.rst [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+* a title to very briefly describe the specification. This will be shown in traceability tables.
+* an id to uniquely identify the specification. The id naming convention is ``SWDD_<COMPONENT_NAME>-<NUMBER>``
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst]: Give hint, how one can determine the next "free ID".

File: docs/getting_started/start_new_component.rst [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+* free text to describe the specification in more details
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst]: add "." at the end of the sentence.

File: docs/getting_started/start_new_component.rst [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+The component tests are written in GTest and are located in the ``test`` directory.
+
+Here are the main features supporting component tests:
+
+* test cases are written with `GoogleTest <https://google.github.io/googletest/>`_
+* component external interfaces are automatically mocked using Google Mock
+* include/exclude content based on configuration 
+* use `sphinx-needs <https://www.sphinx-needs.com/>`_ to create traceability between documentation, code and tests
+
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst:143]: for all the bullet point above: add "."

File: docs/getting_started/start_new_component.rst [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+        // test code
+    }
+
+The `test` is used to tag the component test case. It shall have:
+
+* a title required to link the test case to the test result. ❗The title must match exactly the test case name.
+* an ``id`` to uniquely identify the test case. The id naming convention is ``TS_<COMPONENT_NAME>-<NUMBER>``
+* a ``tests`` tag to link the test case to the specification. One can link multiple specifications by separating them with a comma.
+* free text to describe the test case in more details
+
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst:171]: for all the bullet point above: add "."

File: docs/getting_started/start_new_component.rst [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+Implementing the Component
+**************************
+
+The component implementation is located in the ``src`` directory.
+
+Here are the main features supporting component implementation:
+
+* use `sphinx-needs <https://www.sphinx-needs.com/>`_ to create traceability between documentation, code and tests
+* include/exclude content based on configuration
+
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst:306]: for all the bullet point above: add "."

File: docs/getting_started/start_new_component.rst [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+        ...
+    }
+
+The ``impl`` is used to tag the component implementation. It shall have:
+
+* a title to very briefly describe the specification. This will be shown in traceability tables.
+* an id to uniquely identify the specification. The id naming convention is ``SWIMPL<_<COMPONENT_NAME>-<NUMBER>``
+* an ``implements`` tag to link the implementation to the specification. One can link multiple specifications by separating them with a comma.
+* free text to describe the specification in more details
+
```
Comment (jomal77) [inline docs/getting_started/start_new_component.rst:331]: for all the bullet point above: add "."

**Review by jomal77 — COMMENTED** [review]
Very good description!
I added some review comments. Kindly check them and decide whether to consider or not.

File: docs/getting_started/start_new_component.rst (hunk 1 above)
Comment (xxthunder) [reply docs/getting_started/start_new_component.rst]:
```suggestion
- configure **all** component artifacts (content will be included/excluded based on configuration)
```

File: docs/getting_started/start_new_component.rst [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+**********
+
+The component tests are written in GTest and are located in the ``test`` directory.
+
+Here are the main features supporting component tests:
+
+* test cases are written with `GoogleTest <https://google.github.io/googletest/>`_
+* component external interfaces are automatically mocked using Google Mock
+* include/exclude content based on configuration 
+* use `sphinx-needs <https://www.sphinx-needs.com/>`_ to create traceability between documentation, code and tests
```
Comment (xxthunder) [inline docs/getting_started/start_new_component.rst]:
```suggestion
* Test cases are written with `GoogleTest <https://google.github.io/googletest/>`_.
* Component external interfaces are automatically mocked using Google Mock.
* Include/exclude content based on configuration.
* Use `sphinx-needs <https://www.sphinx-needs.com/>`_ to create traceability between documentation, code and tests.
```

File: docs/getting_started/start_new_component.rst [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+        CREATE_MOCK(mymock);
+        // test code
+    }
+
+The `test` is used to tag the component test case. It shall have:
+
+* a title required to link the test case to the test result. ❗The title must match exactly the test case name.
+* an ``id`` to uniquely identify the test case. The id naming convention is ``TS_<COMPONENT_NAME>-<NUMBER>``
+* a ``tests`` tag to link the test case to the specification. One can link multiple specifications by separating them with a comma.
+* free text to describe the test case in more details
```
Comment (xxthunder) [inline docs/getting_started/start_new_component.rst]:
```suggestion
* a title required to link the test case to the test result. ❗The title must match exactly the test case name.
* an ``id`` to uniquely identify the test case. The id naming convention is ``TS_<COMPONENT_NAME>-<NUMBER>``
* a ``tests`` tag to link the test case to the specification. One can link multiple specifications by separating them with a comma.
* free text to describe the test case in more details.
```

**Review by xxthunder — CHANGES_REQUESTED** [review]
🦆

File: docs/getting_started/start_new_component.rst (hunk 3 above)
Comment (cuinixam) [reply docs/getting_started/start_new_component.rst]: We should have a dedicated chapter where the component report feature is explained. There we can describe the traceability and how to keep the IDs unique and have a numbering scheme.

**Pull request merged**